src/test/java/uk/gov/ch/oauth/nonce/NonceGeneratorTest.java

```diff
@@ -66,12 +65,11 @@ void doesNotErrorWhenNonceIsNull() {
     @Test
     @DisplayName("Generate Nonce creates unique values, brute force test")
     void generateNonceGeneratesUniqueValues() {
-        List<String> nonces = new ArrayList<>();
-        for (int i = 0; i < 100; i++) {
+        Set<String> nonces = new TreeSet<>();
+        final int iterations = 100;
+        for (int i = 0; i < iterations; i++) {
             String newNonce = nonceGenerator.generateNonce();
-            System.out.println(newNonce);
-            assertFalse(nonces.contains(newNonce));
-            nonces.add(newNonce);
```

Comment:
I don't see it being added to the Set...
I should have also been more explicit in my prior comment, you can do `assertTrue(nonces.add(newNonce));` as you go along.
The performance improvement that you mentioned will be more significant the larger the number of elements added - maybe you could up `iterations` to 1000

> public boolean add(E e)
> Adds the specified element to this set if it is not already present. More formally, adds the specified element e to this set if the set contains no element e2 such that (e==null ? e2==null : e.equals(e2)). If this set already contains the element, the call leaves the set unchanged and returns false.
>

Reply:
I don't know why this was raised against an outdated version of the code.
I realised that I could do assertTrue however in order to save those precious milliseconds I dropped the check.